### PR TITLE
Scaffold scripts

### PR DESCRIPTION
**Pull request by jnhooper**
tl;dr;
----
`npm run review`

summary
---------
make a scaffolding script that will auto update the relevant data.json files so the next app can automagically have new reviews

nitty gritty
----------
once you're done watching/reading something and you're ready to write up a review of it you can now run `npm run review` in your terminal (in this directory). This will start a very small series of questions like "what medium did you consume" and "what was the title" based on these answers the script will add create the appropriate object and add it to the correct data.json (`data/movies/data.json`, `data/books/data.json`, or ` data/tv/data.json`)  it will also take the title of the thing you just finished, snake case it and create a directory, inserting an empty README.md into it. This README file is where you can write down your review. 

proposed workflow
-------------------
1. watch/read something
2. make a branch with that as the title
3. `npm run review` and fill out the prompt
4. start writing, commit it to that branch if you're not quite happy with it yet
5. once happy merge it into main and push it up.

follow up work
--------------
https://github.com/mattlock12/movies/pull/4
this pr abstracts away the `movies` page and some of the nextjs ssg stuff and creates books and tv routes

### CODE REVIEW

**Comment by mattlock12** [comment]
@jnhooper can you explain this to me?

**Comment by mattlock12** [comment]
@jnhooper this sounds awesome. Can we work in the concept of `unfinished` to this? Either in the review wizard or maybe just have it push them to an incomplete/ folder by default?

File: scaffold/review.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const inquirer = require('inquirer');
+const writeData = require('./writeData.js')
```
Comment (mattlock12) [inline scaffold/review.js:2]: We have to use commonjs module requirements?

File: scaffold/writeData.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const jsonPath = path.join(mediumPath, 'data.json')
+
+
+  let data  = fs.readFileSync(jsonPath, 'utf8');
+  data = JSON.parse(data);
+
+  if(find(data, (art) => trimEnd(art.title.toLowerCase()) ===trimEnd(answers.title.toLowerCase()))){
+    console.error(`${answers.medium}: "${answers.title}" already exists. consider updating that review`)
+  }
+  else{
```
Comment (mattlock12) [inline scaffold/writeData.js:18]: Weird formatting

File: scaffold/writeData.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    const dirname = snakeCase(answers.title)
+    data.push({
+      title: answers.title,
+      id: data.length + 1,
```
Comment (mattlock12) [inline scaffold/writeData.js:22]: Are you a caveman?

**Comment by mattlock12** [comment]
Just a couple nits but :shipit: as far as I'm concerned

**Review by mattlock12 — APPROVED** [review]

File: scaffold/review.js (hunk 1 above)
Comment (jnhooper) [reply scaffold/review.js:2]: i _think_ so because idk if node automatically has this or not. we can try it with import and see if it works. might need to lock the node version somewhere

File: scaffold/writeData.js (hunk 2 above)
Comment (jnhooper) [reply scaffold/writeData.js:18]: you're werid

**Comment by jnhooper** [comment]
> @jnhooper this sounds awesome. Can we work in the concept of `unfinished` to this? Either in the review wizard or maybe just have it push them to an incomplete/ folder by default?

hmmm... maybe we add a flag to the json? i'm thinking the benefit of this is it automates away the file management stuff and updating json files, where if you put it in incomplete, then we need another script to move it out and manage that data stuff. doable but more work. i guess is your flow that you browse the incomplete folder and then decide which one to polish? or do you come in saying "yea i want to finish my review of idiocracy!"
